docs/06_how-to-guides/30_key-value-api/kv_table/20_how-to-create-indexes-kv-table.md

```diff
@@ -115,7 +115,7 @@ class [[eosio::contract]] smrtcontract : public contract {
 
 The property used for the second parameter must be of template type `std::tuple`. The first parameter must be the type of the property indexed non-uniquely, in our case the type `std::string` is used because `first_name` is the property indexed non-uniquely. And the last parameter of the tuple type must be the type of a property name which is unique. In our case the type `eosio::name` is used because property `account_name` is unique. Multiple properties can be indexed non-uniquely as well. In this case the first parameters types correspond to the properties being indexed. And, as previously already mentioned, the last parameter correspond to the type of a property name which is unique.
 
-Refer to the following possible implementation of a non-unique index on property `account_name` using macro `KV_NAMED_INDEX`:
+Refer to the following reference implementation of a non-unique index on property `account_name` using macro `KV_NAMED_INDEX`:
 
 `smartcontract.hpp file`
 
```

Comment:
in this code sample (I think it's this one - the non-unique index using eosio::kv::table::index template class) should the example use last_name_idx not last_name ??? i.e. 

address_table(eosio::name contract_name) {
      init(contract_name, last_name)
    }

Reply:
+ I am also confused in teh KV_NAMED_INDEX("accname"_n, first_name) examples ... should the name of the index be passed to the init here? "accname" not first_name??

Reply:
so to summarise I don't understand from the docs how the init is being used and I'm not sure the parameters passed to the init match the descriptions

Reply:
specifically the name of the index as the second parameter + the contract name as the first parameter.

Reply:
> * I am also confused in teh KV_NAMED_INDEX("accname"_n, first_name) examples ... should the name of the index be passed to the init here? "accname" not first_name??

yes the second param should be `first_name` the column used in the KV_NAMED_INDEX declaration.

Reply:
> in this code sample (I think it's this one - the non-unique index using eosio::kv::table::index template class) should the example use last_name_idx not last_name ??? i.e.
> 
> address_table(eosio::name contract_name) {
> init(contract_name, last_name)
> }

indeed this should be last_name_idx

I added a step 4 which was missing for 3 of the sub-sections in which it explains wha this second param should be.
this second param of init(...) is different based on how the index was created, via macro KV_NAMED_INDEX or via direct table index template declaration/definition.